sdk/core/core/src/az_aad.c

```diff
@@ -81,11 +81,17 @@ AZ_NODISCARD az_result _az_aad_build_body(
   return AZ_OK;
 }
 
-AZ_NODISCARD az_result _az_aad_request_token(az_http_request * ref_request, _az_token * out_token) {
-  AZ_RETURN_IF_FAILED(az_http_request_append_header(
+AZ_NODISCARD az_result _az_aad_request_token(
+    az_http_client_fn * http_client,
```

Comment:
After you do what I say below (see next comment), then this first parameter should be a pointer to a transport_policy_options structure.